### PR TITLE
[feedly] Add a Data Connector to pull IoCs from Feedly

### PR DESCRIPTION
**Pull request by Mathieu4141**
Change(s):
 - Adding a Data Connector to ingest IoCs from Feedly

 Reason for Change(s):
 - New feature

 Version Updated:
 - No

 Testing Completed:
 - Yes

 Checked that the validations are passing and have addressed any issues that are present:
 - Need help (see logs below)

<details>
  <summary>KqlvalidationsTests logs</summary>
  
```bash
Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
[xUnit.net 00:00:00.82] Exception discovering tests from Kqlvalidations.Tests: System.BadImageFormatException: Could not load file or assembly 'Microsoft.Azure.Sentinel.KustoServices, Version=6.5.0.0, Culture=neutral, PublicKeyToken=null'. An attempt was made to load a program with an incorrect format.

File name: 'Microsoft.Azure.Sentinel.KustoServices, Version=6.5.0.0, Culture=neutral, PublicKeyToken=null'
   at System.Reflection.RuntimeAssembly.GetExportedTypes()
   at System.Reflection.Assembly.get_ExportedTypes()
   at Xunit.Sdk.ReflectionAssemblyInfo.GetTypes(Boolean includePrivateTypes) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Reflection\ReflectionAssemblyInfo.cs:line 80
   at Xunit.Sdk.TestCollectionFactoryHelper.GetTestCollectionDefinitions(IAssemblyInfo assemblyInfo, IMessageSink diagnosticMessageSink) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\TestCollectionFactoryHelper.cs:line 22
   at Xunit.Sdk.CollectionPerClassTestCollectionFactory..ctor(ITestAssembly testAssembly, IMessageSink diagnosticMessageSink) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\CollectionPerClassTestCollectionFactory.cs:line 27
   at Xunit.Sdk.ExtensibilityPointFactory.GetXunitTestCollectionFactory(IMessageSink diagnosticMessageSink, IAttributeInfo collectionBehaviorAttribute, ITestAssembly testAssembly) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\ExtensibilityPointFactory.cs:line 239
   at Xunit.Sdk.XunitTestFrameworkDiscoverer..ctor(IAssemblyInfo assemblyInfo, ISourceInformationProvider sourceProvider, IMessageSink diagnosticMessageSink, IXunitTestCollectionFactory collectionFactory) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\XunitTestFrameworkDiscoverer.cs:line 45
   at Xunit.Sdk.XunitTestFramework.CreateDiscoverer(IAssemblyInfo assemblyInfo) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\XunitTestFramework.cs:line 21
   at Xunit.Sdk.TestFramework.GetDiscoverer(IAssemblyInfo assemblyInfo) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\TestFramework.cs:line 71
   at Xunit.Sdk.TestFrameworkProxy.GetDiscoverer(IAssemblyInfo assembly) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\TestFrameworkProxy.cs:line 67
   at Xunit.Xunit2Discoverer..ctor(AppDomainSupport appDomainSupport, ISourceInformationProvider sourceInformationProvider, IAssemblyInfo assemblyInfo, String assemblyFileName, String xunitExecutionAssemblyPath, String configFileName, Boolean shadowCopy, String shadowCopyFolder, IMessageSink diagnosticMessageSink, Boolean verifyAssembliesOnDisk) in C:\Dev\xunit\xunit\src\xunit.runner.utility\Frameworks\v2\Xunit2Discoverer.cs:line 106
   at Xunit.Xunit2Discoverer..ctor(AppDomainSupport appDomainSupport, ISourceInformationProvider sourceInformationProvider, String assemblyFileName, String configFileName, Boolean shadowCopy, String shadowCopyFolder, IMessageSink diagnosticMessageSink, Boolean verifyAssembliesOnDisk) in C:\Dev\xunit\xunit\src\xunit.runner.utility\Frameworks\v2\Xunit2Discoverer.cs:line 62
   at Xunit.Xunit2..ctor(AppDomainSupport appDomainSupport, ISourceInformationProvider sourceInformationProvider, String assemblyFileName, String configFileName, Boolean shadowCopy, String shadowCopyFolder, IMessageSink diagnosticMessageSink, Boolean verifyTestAssemblyExists) in C:\Dev\xunit\xunit\src\xunit.runner.utility\Frameworks\v2\Xunit2.cs:line 41
   at Xunit.XunitFrontController.CreateInnerController() in C:\Dev\xunit\xunit\src\xunit.runner.utility\Frameworks\XunitFrontController.cs:line 144
   at Xunit.XunitFrontController.EnsureInitialized() in C:\Dev\xunit\xunit\src\xunit.runner.utility\Frameworks\XunitFrontController.cs:line 163
   at Xunit.XunitFrontController.get_InnerController() in C:\Dev\xunit\xunit\src\xunit.runner.utility\Frameworks\XunitFrontController.cs:line 103
   at Xunit.XunitFrontController.get_TargetFramework() in C:\Dev\xunit\xunit\src\xunit.runner.utility\Frameworks\XunitFrontController.cs:line 111
   at Xunit.Runner.VisualStudio.VsTestRunner.DiscoverTestsInSource[TVisitor](XunitFrontController framework, LoggerHelper logger, TestPlatformContext testPlatformContext, RunSettings runSettings, Func`4 visitorFactory, Action`4 visitComplete, String assemblyFileName, Boolean shadowCopy, TestAssemblyConfiguration configuration) in /_/src/xunit.runner.visualstudio/VsTestRunner.cs:line 283
No test is available in /Users/cytadel/feedly/Azure-Sentinel/.script/tests/KqlvalidationsTests/bin/Debug/net6.0/Kqlvalidations.Tests.dll. Make sure that test discoverer & executors are registered and platform & framework version settings are appropriate and try again.

Additionally, path to test adapters can be specified using /TestAdapterPath command. Example  /TestAdapterPath:<pathToCustomAdapters>.

```
  
</details>

### CODE REVIEW

**Comment by Mathieu4141** [comment]
@microsoft-github-policy-service agree company="Feedly"

**Comment by v-prasadboke** [comment]
Hello @Mathieu4141, Thank you for raising this PR. 
I'll examine this PR and Update you about the same till 07 Aug 2023. 
Thanks!!

**Comment by v-prasadboke** [comment]
Hello @Mathieu4141, Function app zip should be included in the data connector folder.
Please go through this readme file for more clarification  https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Connector_REST_API_AzureFunctionApp_template/README.md

**Comment by v-prasadboke** [comment]
Hello @Mathieu4141, Thank you for committing the changes. Continuing to review this PR. 
Thanks!

File: Solutions/Feedly/Data Connectors/azuredeploy_Connector_Feedly_AzureFunction.json
Comment (v-prasadboke) [inline Solutions/Feedly/Data Connectors/azuredeploy_Connector_Feedly_AzureFunction.json:1]: Hello @Mathieu4141, a parameter is missing for WorkspaceID. Please add this parameter in global parameter section of this azure deploy file.

File: Solutions/Feedly/SolutionMetadata.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	"offerId": "azure-sentinel-solution-feedly",
+	"firstPublishDate": "2023-08-01",
+	"providers": [ "Feedly" ],
+	"categories": {
+		"domains": ["Security - Threat Protection"]
+	},
+	"support": {
+		"name": "Feedly Inc",
+		"email": "ml@feedly.com",
+		"tier": "Feedly",
```
Comment (v-prasadboke) [inline Solutions/Feedly/SolutionMetadata.json:12]: Please change tier to Partner.

File: Solutions/Feedly/Data/Solution_Feedly.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "Name": "Feedly",
+  "Author": "Feedly ml@feedly.com",
+  "Logo": "<img src=\"https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Logos/Feedly.svg\" width=\"75px\" height=\"75px\">",
+  "Description": "TODO",
+  "Data Connectors": ["Solutions/Feedly/Data Connectors/Feedly_API_AzureFunctionApp.json"],
+  "BasePath": "C:\\GitHub\\Azure-Sentinel\\Solutions\\Feedly",
+  "Version": "1.0.0",
```
Comment (v-prasadboke) [inline Solutions/Feedly/Data/Solution_Feedly.json:8]: Change the Version to 3.0.0

**Comment by v-prasadboke** [comment]
After making this changes please repackage the solution again. Thank you.

**Comment by v-prasadboke** [comment]
Also I've added the release notes file please update the date modified section of the release notes as per needed.

File: Solutions/Feedly/Data/Solution_Feedly.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+  "Name": "Feedly",
+  "Author": "Feedly ml@feedly.com",
+  "Logo": "<img src=\"https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Logos/Feedly.svg\" width=\"75px\" height=\"75px\">",
+  "Description": "TODO",
```
Comment (v-prasadboke) [inline Solutions/Feedly/Data/Solution_Feedly.json:5]: Please add atleast a short description of 2 to 3 lines of the solution.

**Comment by v-prasadboke** [comment]
Other than this all looks good. Thanks, Prasad Boke.

**Comment by Mathieu4141** [comment]
Thank for the review @v-prasadboke !

I made the changes and re-packaged the solution

**Comment by v-prasadboke** [comment]
Hello @Mathieu4141, Thank you for committing the changes. I'll resume my investigation and update you by 223 Aug 2023.

**Comment by v-prasadboke** [comment]
Hello @Mathieu4141, A parameter **WorkspaceID** in azuredeploy file of data connector is still not in used. Please declare it in global parameter section of the code.
image

**Comment by Mathieu4141** [comment]
Hi @v-prasadboke

I used `SentinelWorkspaceId` to avoid confusion with a potential feedly workspace id (I saw a few other solutions that used `AzureSentinelWorkspaceId`), but I can change to `WorkspaceID` everywhere if that doesn't work or isn't good practice

I may be looking in the wrong place, but I'm seing `SentinelWorkspaceId` and not `WorkspaceID` in the azuredeploy file of the connector:
https://github.com/Azure/Azure-Sentinel/blob/299bc571082155658f5179d093714b1d803d402f/Solutions/Feedly/Data%20Connectors/azuredeploy_Connector_Feedly_AzureFunction.json#L47

**Review by v-prasadboke — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**